Name can't be set via programmatic configuration

Hello everyone!

I'm trying to use hazelcast-eureka-one and hazelcast only programmatically, since I'm using Spring for my DI with the hazelcast-spring.jar, it's enough to configure the appropriate Config Bean and Spring would create a hazelcast instance for me. But whatever I do, I can't seem to get the name of the eureka client instance right.

My setup is: Maven Dependencies
```xml
<dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-cache</artifactId>
            <version>2.0.1.RELEASE</version>
        </dependency>
        <dependency>
            <groupId>com.hazelcast</groupId>
            <artifactId>hazelcast</artifactId>
            <version>3.9.3</version>
        </dependency>
        <dependency>
            <groupId>com.hazelcast</groupId>
            <artifactId>hazelcast-spring</artifactId>
            <version>3.9.3</version>
        </dependency>
        <dependency>
            <groupId>com.hazelcast</groupId>
            <artifactId>hazelcast-eureka-one</artifactId>
            <version>1.0.2</version>
        </dependency>
    </dependencies>
```

With the following Config Bean configuration:

```kotlin
@Configuration
class CacheConfiguration {
    @Bean
    fun configuration(
            @Value("\${eureka.client.serviceUrl.defaultZone:http://127.0.0.1:8761/eureka/}") eurekaServiceUrl: String): Config {
        val config = Config()
        val mapConfig = MapConfig("priorities")
                .setTimeToLiveSeconds(10 * 60)
                .setEvictionPolicy(EvictionPolicy.LRU)
                .setMaxSizeConfig(MaxSizeConfig(10, MaxSizeConfig.MaxSizePolicy.FREE_HEAP_SIZE))
        config.addMapConfig(mapConfig)

        val networkConfig = NetworkConfig()
        val joinConfig = JoinConfig()
        val awsConfig = AwsConfig()
        awsConfig.isEnabled = false
        joinConfig.awsConfig = awsConfig
        val tcpIpConfig = TcpIpConfig()
        tcpIpConfig.isEnabled = false
        joinConfig.tcpIpConfig = tcpIpConfig
        val multicastConfig = MulticastConfig()
        multicastConfig.isEnabled = false
        joinConfig.multicastConfig = multicastConfig

        val eurekaDiscoveryConfig = DiscoveryConfig()
        val eurekaDiscoveryStrategyConfig = DiscoveryStrategyConfig(com.hazelcast.eureka.one.EurekaOneDiscoveryStrategyFactory(),
                mapOf(
                        "use-classpath-eureka-client-props" to false,
                        "shouldUseDns" to false,
                        "name" to "hazelcast-cache-cluster",
                        "serviceUrl.default" to eurekaServiceUrl
                ) as Map<String, Comparable<Any?>>?)
        eurekaDiscoveryConfig.discoveryStrategyConfigs.add(eurekaDiscoveryStrategyConfig)
        
        joinConfig.discoveryConfig = eurekaDiscoveryConfig
        networkConfig.join = joinConfig
        config.networkConfig = networkConfig

        config.setProperty("hazelcast.discovery.enabled", "true")

        return config
    }
}
```

So this setup basically works fine, its just that the name of the eureka instance is always UNKNOWN.
When i have eureka-client.properties in place, it just works as expected and the name of the eureka client instance is set correctly.

```properties
hazelcast.shouldUseDns=false
hazelcast.name=hazelcast-cache-cluster
hazelcast.serviceUrl.default=http://127.0.0.1:8761/eureka/
```

Now i debugged through this a little, and the cause seems to be that the name property gets filtered out as to when it gets passed down to the discovery strategy, and I also now know that the file based approach works cause when `MyDataCenterInstanceConfig` is instantiated, it passes the namespace to the `PropertiesInstanceConfig` which in turn uses the `Archaius1Utils.initConfig(CommonConstants.CONFIG_FILE_NAME)` method call to search for the properties file "eureka-client" on the classpath and read the data.

The problem for me now is that I would also like to set a name programmatically, but I really have no clue on how this should be fixed, or even if it is supposed to be this way?

Anyways, thanks for any help in advance!

Hello. Not sure if this is good place to write about my issue. I am using `hazelcast-eureka plugin`. I get exception when trying to provide properties to DiscoveryStrategyConfig programaticaly using java config like this:
```
                final Config config = new Config();
		config.setInstanceName(instanceName);
		config.setProperty("hazelcast.discovery.enabled", Boolean.TRUE.toString());

		final DiscoveryStrategyConfig strategyConfig = new DiscoveryStrategyConfig(new EurekaOneDiscoveryStrategyFactory());
		strategyConfig.addProperty("use-classpath-eureka-client-props", false);
               //more props here
		final JoinConfig join = config.getNetworkConfig().getJoin();
		join.getMulticastConfig().setEnabled(false);
		join.getDiscoveryConfig().addDiscoveryStrategyConfig(strategyConfig);
		return config;
```
Exception:

> Caused by: java.lang.IllegalStateException: ConfigurationManager is already initialized with configuration org.apache.commons.configuration.MapConfiguration@4c78e00c

Line `strategyConfig.addProperty("use-classpath-eureka-client-props", false);` causes it. 

I am using `hazelcast-eureka-one 1.0.2` and `hazelcast:3.7.8`. It works correct when I am using `eureka-client.properties` file.

Is there any way to configure eureka plugin programmatically?

Full trace:

> org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'hazelcastInstance' defined in class path resource [com/rbc/gam/head2head/config/HazelcastConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.hazelcast.core.HazelcastInstance]: Factory method 'hazelcastInstance' threw exception; nested exception is java.lang.RuntimeException: Failed to configure discovery strategies
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:599)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1178)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1072)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:511)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:481)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:312)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:308)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:297)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1080)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:857)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:543)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:124)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107)
	at com.rbc.gam.head2head.Head2headApplication.main(Head2headApplication.java:18)
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.hazelcast.core.HazelcastInstance]: Factory method 'hazelcastInstance' threw exception; nested exception is java.lang.RuntimeException: Failed to configure discovery strategies
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:189)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588)
	... 20 common frames omitted
Caused by: java.lang.RuntimeException: Failed to configure discovery strategies
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.loadDiscoveryStrategies(DefaultDiscoveryService.java:168)
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.<init>(DefaultDiscoveryService.java:60)
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryServiceProvider.newDiscoveryService(DefaultDiscoveryServiceProvider.java:29)
	at com.hazelcast.instance.Node.createDiscoveryService(Node.java:226)
	at com.hazelcast.instance.Node.<init>(Node.java:193)
	at com.hazelcast.instance.HazelcastInstanceImpl.createNode(HazelcastInstanceImpl.java:153)
	at com.hazelcast.instance.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:125)
	at com.hazelcast.instance.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:218)
	at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:176)
	at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:126)
	at com.hazelcast.core.Hazelcast.newHazelcastInstance(Hazelcast.java:58)
	at com.rbc.gam.head2head.config.HazelcastConfig.hazelcastInstance(HazelcastConfig.java:51)
	at com.rbc.gam.head2head.config.HazelcastConfig$$EnhancerBySpringCGLIB$$d1406f60.CGLIB$hazelcastInstance$1(<generated>)
	at com.rbc.gam.head2head.config.HazelcastConfig$$EnhancerBySpringCGLIB$$d1406f60$$FastClassBySpringCGLIB$$c125b230.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:358)
	at com.rbc.gam.head2head.config.HazelcastConfig$$EnhancerBySpringCGLIB$$d1406f60.hazelcastInstance(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
	... 21 common frames omitted
Caused by: java.lang.IllegalStateException: ConfigurationManager is already initialized with configuration org.apache.commons.configuration.MapConfiguration@4c78e00c
	at com.netflix.config.DynamicPropertyFactory.initWithConfigurationSource(DynamicPropertyFactory.java:146)
	at com.hazelcast.eureka.one.PropertyBasedEurekaClientConfig.<init>(PropertyBasedEurekaClientConfig.java:57)
	at com.hazelcast.eureka.one.EurekaOneDiscoveryStrategy.<init>(EurekaOneDiscoveryStrategy.java:179)
	at com.hazelcast.eureka.one.EurekaOneDiscoveryStrategy.<init>(EurekaOneDiscoveryStrategy.java:63)
	at com.hazelcast.eureka.one.EurekaOneDiscoveryStrategy$EurekaOneDiscoveryStrategyBuilder.build(EurekaOneDiscoveryStrategy.java:125)
	at com.hazelcast.eureka.one.EurekaOneDiscoveryStrategyFactory.newDiscoveryStrategy(EurekaOneDiscoveryStrategyFactory.java:57)
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.buildDiscoveryStrategy(DefaultDiscoveryService.java:217)
	at com.hazelcast.spi.discovery.impl.DefaultDiscoveryService.loadDiscoveryStrategies(DefaultDiscoveryService.java:158)
	... 42 common frames omitted

Any update on this bug? Facing it too for `hazelcast versions 3.9-3.11.1` and `hazelcast-eureka-one 1.0.2 - 1.1`. When using lower hazelcast vesrion - it does not let me set properties programmatically  from Java code. 
For `hazelcast 3.11.1` I get next exception:

> Caused by: com.hazelcast.config.InvalidConfigurationException: Unknown properties: '[name]' on discovery strategy

--- a/src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java
+++ b/src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java
@@ -58,6 +58,7 @@ import static com.hazelcast.eureka.one.EurekaOneProperties.NAMESPACE;
 import static com.hazelcast.eureka.one.EurekaOneProperties.SELF_REGISTRATION;
 import static com.hazelcast.eureka.one.EurekaOneProperties.SKIP_EUREKA_REGISTRATION_VERIFICATION;
 import static com.hazelcast.eureka.one.EurekaOneProperties.USE_CLASSPATH_EUREKA_CLIENT_PROPS;
+import static com.hazelcast.eureka.one.EurekaOneProperties.NAME;
 import static com.hazelcast.eureka.one.EurekaOneProperties.USE_METADATA_FOR_HOST_AND_PORT;
 
 final class EurekaOneDiscoveryStrategy
@@ -184,6 +185,11 @@ final class EurekaOneDiscoveryStrategy
         }
     }
 
+    private String getAppname() {
+        Comparable name = this.getProperties().get(NAME.key());
+        return name == null ? "unknown" : name.toString();
+    }
+
     private Map<String, Object> getEurekaClientProperties(String namespace, Map<String, Comparable> properties) {
         Map<String, Object> result = new HashMap<String, Object>();
         for (Map.Entry<String, Comparable> e : properties.entrySet()) {
@@ -230,7 +236,12 @@ final class EurekaOneDiscoveryStrategy
             if ("cloud".equals(value.trim().toLowerCase())) {
                 return new DelegatingInstanceConfig(new CloudInstanceConfig(this.namespace), localNode);
             }
-            return new DelegatingInstanceConfig(new MyDataCenterInstanceConfig(this.namespace), localNode);
+            if (this.useClasspathEurekaClientProps) {
+                return new DelegatingInstanceConfig(new MyDataCenterInstanceConfig(this.namespace), localNode);
+            }
+            else {
+                return new DelegatingInstanceConfig(new MyDataCenterInstanceConfig(this.namespace), localNode, getAppname());
+            }
         } catch (IOException e) {
             throw new IllegalStateException("Cannot build EurekaInstanceInfo", e);
         }
@@ -389,11 +400,20 @@ final class EurekaOneDiscoveryStrategy
         private final EurekaInstanceConfig instanceConfig;
         private final DiscoveryNode localNode;
         private final String uuid;
+        private final String appname;
 
         private DelegatingInstanceConfig(EurekaInstanceConfig instanceConfig, DiscoveryNode localNode) {
             this.instanceConfig = instanceConfig;
             this.localNode = localNode;
             this.uuid = UuidUtil.newSecureUuidString();
+            this.appname = instanceConfig.getAppname();
+        }
+
+        private DelegatingInstanceConfig(EurekaInstanceConfig instanceConfig, DiscoveryNode localNode, String appname) {
+            this.instanceConfig = instanceConfig;
+            this.localNode = localNode;
+            this.uuid = UuidUtil.newSecureUuidString();
+            this.appname = appname;
         }
 
         public String getInstanceId() {
@@ -401,7 +421,7 @@ final class EurekaOneDiscoveryStrategy
         }
 
         public String getAppname() {
-            return instanceConfig.getAppname();
+            return appname;
         }
 
         public String getAppGroupName() {

--- a/src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java
+++ b/src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java
@@ -52,14 +52,7 @@ import java.util.Map;
 import java.util.Properties;
 import java.util.concurrent.TimeUnit;
 
-import static com.hazelcast.eureka.one.EurekaOneProperties.EUREKA_ONE_SYSTEM_PREFIX;
-import static com.hazelcast.eureka.one.EurekaOneProperties.HZ_PROPERTY_DEFINITIONS;
-import static com.hazelcast.eureka.one.EurekaOneProperties.NAMESPACE;
-import static com.hazelcast.eureka.one.EurekaOneProperties.SELF_REGISTRATION;
-import static com.hazelcast.eureka.one.EurekaOneProperties.SKIP_EUREKA_REGISTRATION_VERIFICATION;
-import static com.hazelcast.eureka.one.EurekaOneProperties.USE_CLASSPATH_EUREKA_CLIENT_PROPS;
-import static com.hazelcast.eureka.one.EurekaOneProperties.NAME;
-import static com.hazelcast.eureka.one.EurekaOneProperties.USE_METADATA_FOR_HOST_AND_PORT;
+import static com.hazelcast.eureka.one.EurekaOneProperties.*;
 
 final class EurekaOneDiscoveryStrategy
         extends AbstractDiscoveryStrategy {
@@ -231,7 +224,7 @@ final class EurekaOneDiscoveryStrategy
                 String key = String.format("%s.datacenter", this.namespace);
                 value = props.getProperty(key, "");
             } else {
-                value = String.valueOf(getProperties().get("datacenter"));
+                value = String.valueOf(getProperties().get(DATACENTER.key()));
             }
             if ("cloud".equals(value.trim().toLowerCase())) {
                 return new DelegatingInstanceConfig(new CloudInstanceConfig(this.namespace), localNode);

--- a/src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java
+++ b/src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java
@@ -396,10 +396,7 @@ final class EurekaOneDiscoveryStrategy
         private final String appname;
 
         private DelegatingInstanceConfig(EurekaInstanceConfig instanceConfig, DiscoveryNode localNode) {
-            this.instanceConfig = instanceConfig;
-            this.localNode = localNode;
-            this.uuid = UuidUtil.newSecureUuidString();
-            this.appname = instanceConfig.getAppname();
+            this(instanceConfig, localNode, instanceConfig.getAppname());
         }
 
         private DelegatingInstanceConfig(EurekaInstanceConfig instanceConfig, DiscoveryNode localNode, String appname) {

--- a/src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java
+++ b/src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java
@@ -52,7 +52,15 @@ import java.util.Map;
 import java.util.Properties;
 import java.util.concurrent.TimeUnit;
 
-import static com.hazelcast.eureka.one.EurekaOneProperties.*;
+import static com.hazelcast.eureka.one.EurekaOneProperties.DATACENTER;
+import static com.hazelcast.eureka.one.EurekaOneProperties.EUREKA_ONE_SYSTEM_PREFIX;
+import static com.hazelcast.eureka.one.EurekaOneProperties.HZ_PROPERTY_DEFINITIONS;
+import static com.hazelcast.eureka.one.EurekaOneProperties.NAME;
+import static com.hazelcast.eureka.one.EurekaOneProperties.NAMESPACE;
+import static com.hazelcast.eureka.one.EurekaOneProperties.SELF_REGISTRATION;
+import static com.hazelcast.eureka.one.EurekaOneProperties.SKIP_EUREKA_REGISTRATION_VERIFICATION;
+import static com.hazelcast.eureka.one.EurekaOneProperties.USE_CLASSPATH_EUREKA_CLIENT_PROPS;
+import static com.hazelcast.eureka.one.EurekaOneProperties.USE_METADATA_FOR_HOST_AND_PORT;
 
 final class EurekaOneDiscoveryStrategy
         extends AbstractDiscoveryStrategy {
@@ -232,9 +240,7 @@ final class EurekaOneDiscoveryStrategy
             if (this.useClasspathEurekaClientProps) {
                 return new DelegatingInstanceConfig(new MyDataCenterInstanceConfig(this.namespace), localNode);
             }
-            else {
-                return new DelegatingInstanceConfig(new MyDataCenterInstanceConfig(this.namespace), localNode, getAppname());
-            }
+            return new DelegatingInstanceConfig(new MyDataCenterInstanceConfig(this.namespace), localNode, getAppname());
         } catch (IOException e) {
             throw new IllegalStateException("Cannot build EurekaInstanceInfo", e);
         }

--- a/src/main/java/com/hazelcast/eureka/one/EurekaOneProperties.java
+++ b/src/main/java/com/hazelcast/eureka/one/EurekaOneProperties.java
@@ -61,6 +61,15 @@ public final class EurekaOneProperties {
             property("use-classpath-eureka-client-props", BOOLEAN);
 
     /**
+     * <p>Configuration key: <tt>name</tt></p>
+     * <p>Defines the App Name.
+     * This is only used when <tt>use-classpath-eureka-client-props</tt> is <tt>false</tt>.</p>
+     * <p>The default value is: <tt>unknown</tt></p>
+     */
+    public static final PropertyDefinition NAME =
+            property("name", STRING);
+
+    /**
      * <p>Configuration key: <tt>self-registration</tt></p>
      * <p>Defines if the Discovery SPI plugin will register itself with the Eureka 1 service discovery.</p>
      * <p>The default value is: <tt>true</tt></p>
@@ -109,6 +118,7 @@ public final class EurekaOneProperties {
 
     static final Collection<PropertyDefinition> HZ_PROPERTY_DEFINITIONS = Lists.newArrayList(
             USE_CLASSPATH_EUREKA_CLIENT_PROPS,
+            NAME,
             SELF_REGISTRATION,
             NAMESPACE,
             USE_METADATA_FOR_HOST_AND_PORT,

--- a/src/main/java/com/hazelcast/eureka/one/EurekaOneProperties.java
+++ b/src/main/java/com/hazelcast/eureka/one/EurekaOneProperties.java
@@ -70,6 +70,14 @@ public final class EurekaOneProperties {
             property("name", STRING);
 
     /**
+     * <p>Configuration key: <tt>datacenter</tt></p>
+     * <p>Defines the Eureka server datacenter.
+     * This is only used when <tt>use-classpath-eureka-client-props</tt> is <tt>false</tt>.</p>
+     */
+    public static final PropertyDefinition DATACENTER =
+            property("datacenter", STRING);
+
+    /**
      * <p>Configuration key: <tt>self-registration</tt></p>
      * <p>Defines if the Discovery SPI plugin will register itself with the Eureka 1 service discovery.</p>
      * <p>The default value is: <tt>true</tt></p>
@@ -119,6 +127,7 @@ public final class EurekaOneProperties {
     static final Collection<PropertyDefinition> HZ_PROPERTY_DEFINITIONS = Lists.newArrayList(
             USE_CLASSPATH_EUREKA_CLIENT_PROPS,
             NAME,
+            DATACENTER,
             SELF_REGISTRATION,
             NAMESPACE,
             USE_METADATA_FOR_HOST_AND_PORT,

--- a/src/test/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategyTest.java
+++ b/src/test/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategyTest.java
@@ -75,7 +75,7 @@ public class EurekaOneDiscoveryStrategyTest extends AbstractEurekaOneDiscoverySt
         InstanceInfo mockInfo = mock(InstanceInfo.class);
         when(mockInfo.getId()).thenReturn(RandomStringUtils.random(42));
         when(mockInfo.getStatus()).thenReturn(InstanceInfo.InstanceStatus.UP);
-        when(mockInfo.getIPAddr()).thenReturn("local");
+        when(mockInfo.getIPAddr()).thenReturn("..."); // invalid name / IP address
 
         application.addInstance(mockInfo);
 

--- a/src/test/java/com/hazelcast/eureka/one/EurekaOnePropertyBasedClientConfigTest.java
+++ b/src/test/java/com/hazelcast/eureka/one/EurekaOnePropertyBasedClientConfigTest.java
@@ -59,5 +59,7 @@ public class EurekaOnePropertyBasedClientConfigTest extends AbstractEurekaOneDis
         assertThat(client.getEurekaClientConfig(), instanceOf(PropertyBasedEurekaClientConfig.class));
         assertThat(client.getEurekaClientConfig().getEurekaServerServiceUrls("default").get(0),
                 is(properties.get("serviceUrl.default")));
+        assertThat(client.getApplicationInfoManager().getEurekaInstanceConfig().getAppname(),
+                is(properties.get("name")));
     }
 }

